cypress/support/fragments/data_import/dataImportViewAllPage.js

```diff
@@ -15,8 +26,239 @@ export default {
   },
 
   checkRowsCount(rowCount) {
-    return cy.expect(MultiColumnList({ id: 'list-data-import' }).has({ rowCount }));
+    if (rowCount === 0) {
+      cy.expect(MultiColumnList({ id: 'list-data-import' }).is({ visible: false }));
+    } else {
+      cy.expect(MultiColumnList({ id: 'list-data-import' }).has({ rowCount }));
+    }
+  },
+
+  checkById({ id }) {
+    const queryString = UrlParams.getSearchByIdQueryString({ id });
+    this.getNumberOfMatchedJobs(queryString).then(count => {
+      this.checkRowsCount(count);
+    });
   },
 
   options: ['Keyword (ID, File name)', 'ID', 'File name'],
+
+  errorsInImportStatuses: ['No', 'Yes'],
+
+  singleRecordImportsStatuses: ['No', 'Yes'],
+
+  resetAllFilters() {
+    cy.do(Button('Reset all').click());
+
+    // After resetting all filters, we need to sort MultiColumnList
+    // Otherwise, server cannot parse request params and returns error with 422 status
+    // In this case, sort by completed date in ascending order
+    cy.do(MultiColumnListHeader('Ended running').click());
+  },
+
+  filterJobsByErrors(filter) {
+    if (filter === 'Yes') {
+      return cy.do(CheckBox({ id: 'clickable-filter-statusAny-error' }).click());
+    } else {
+      return cy.do(CheckBox({ id: 'clickable-filter-statusAny-committed' }).click());
+    }
+  },
+
+  filterJobsByDate({ from, end }) {
+    cy.do([
+      Accordion({ id: 'completedDate' }).clickHeader(),
+      TextField({ label: 'From' }).fillIn(from),
+      TextField({ label: 'To' }).fillIn(end),
+      Button('Apply').click()
+    ]);
+  },
+
+  filterJobsByJobProfile(jobProfile) {
+    cy.do([
+      Accordion({ id: 'profileIdAny' }).clickHeader(),
+      Selection({ value: 'Choose job profile' }).open(),
+      SelectionList().select(jobProfile)
+    ]);
+  },
+
+  filterJobsByUser(user) {
+    cy.do([Accordion({ id: 'userId' }).clickHeader(),
+      Selection({ value: 'Choose user' }).open(),
+      SelectionList().select(user)
+    ]);
+  },
+
+  filterJobsByInventorySingleRecordImports(filter) {
+    if (filter === 'Yes') {
+      cy.do(CheckBox({ id: 'clickable-filter-singleRecordImports-yes' }).click());
+    } else {
+      cy.do(CheckBox({ id: 'clickable-filter-singleRecordImports-no' }).click());
+    }
+  },
+
+  waitUIToBeFiltered() {
```

Comment:
have you tried to work through unique file name? It's common way to apply some filter. Prepare test data with unique special content(generated file with unique name) -> apply special filter -> wait presence of unique string into filtering results

Reply:
@dmitriyfolio, in that case, wouldn't every filter require unique file?

Reply:
yes if it's needed. Check you test please. May be it's not required in some cases